### DISCUSSION TITLE
can buildx bake support --env_file flag like docker-compose for custom .env files

### DISCUSSION THREAD

**Discussion by dkebler** [docker/buildx#1339] [Q&A]
with docker compose you can specify a custom .env file
e.g.
`docker compose --env-file ./config/.env.dev up`
https://docs.docker.com/engine/release-notes/

this doesn't work obviously
`docker buildx bake --env-file my.env --print`
`unknown flag: --env-file`

if this is desired then I can make a issue.

**Comment by crazy-max** [comment]
You can use override configuration file as shown in https://docs.docker.com/build/bake/configuring-build/

```hcl
# docker-bake.hcl
variable "FOO" {
  default = "abc"
}

target "app" {
  args = {
    v1 = "pre-${FOO}"
  }
}
```

```
# env.hcl
WHOAMI="myuser"
FOO="def-${WHOAMI}"
```

```
$ docker buildx bake -f docker-bake.hcl -f env.hcl --print app
{
  "group": {
    "default": {
      "targets": [
        "app"
      ]
    }
  },
  "target": {
    "app": {
      "context": ".",
      "dockerfile": "Dockerfile",
      "args": {
        "v1": "pre-def-myuser"
      }
    }
  }
}
```

You still need to define [`variable` blocks](https://docs.docker.com/build/bake/file-definition/#variable) for env vars you want to use.

**Comment by ravensorb** [comment]
This is a lot more complicated -- any chance of adding the parameter for parity and simplicity sake?
